src/core/components/courses/SlotCreationModal.vue

```diff
@@ -68,22 +69,17 @@ export default {
       this.$emit('submit');
     },
     update (event, prop) {
-      this.$emit('update:newCourseSlot', { ...this.newCourseSlot, [prop]: event });
+      this.$emit('update:newCourseSlot', set(this.newCourseSlot, prop, event));
     },
     getType (step) {
       return step ? this.stepOptions.find(option => option.value === step).type : '';
     },
     updateStep (step) {
       const type = this.getType(step);
-      this.$emit(
-        'update:newCourseSlot',
-        {
-          ...this.newCourseSlot,
-          step,
-          ...(type !== REMOTE && { meetingLink: '' }),
-          ...(type !== ON_SITE && { address: {} }),
-        }
-      );
+      let courseSlot = set(this.newCourseSlot, 'step', step);
+      if (type !== REMOTE) courseSlot = set(courseSlot, 'meetingLink', '');
+      if (type !== ON_SITE) courseSlot = set(courseSlot, 'address', {});
+      this.$emit('update:newCourseSlot', set(courseSlot, 'step', step));
```

Comment:
En fait la methode `set` change l'objet de base donc la partie `let courseSlot = ` est perturbante car en faisant `let courseSlot = set(this.newCourseSlot, 'step', step);` on change dirctement `this.newCourseSlot`
egalement, puisqu'on modifie directement `this.newCourseSlot`, j'ai l'impression qu'on bypasse un peu le ` this.$emit('update:newCourseSlot'`...
je pense qu'il vaut mieux passer par le methode comme dans la PR de Manon
https://lodash.com/docs/4.17.15#set

Reply:
done